Dockerfile

```diff
@@ -1,7 +1,7 @@
 FROM python:3.8
 
 RUN pip install --no-cache-dir --upgrade pip && \
-    pip install --no-cache-dir gradient
```

Comment:
This definitely shouldn't pin. You could add an `ENV VERSION` and set `-e VERSION="$CIRCLECI_TAG` to produce a container with the exactly built version.

Reply:
Gotcha - how does the update look? i removed some pushes and only push on tag two versions

Reply:
i used arg instead of var

Reply:
arg's good